protocol-specs/generators/multisignature_registration/index.js

```diff
@@ -0,0 +1,440 @@
+		createSignatureObject(txBuffer, accounts.mandatoryOne).signature,
+	);
+	tx.signatures.push(
+		createSignatureObject(txBuffer, accounts.optionalOne).signature,
+	);
+	tx.signatures.push(
+		createSignatureObject(txBuffer, accounts.optionalTwo).signature,
+	);
+	txBuffer = serializeMemberSignatures(tx, txBuffer);
+
+	const id = getId(txBuffer);
+
+	tx.id = id;
+
+	return {
+		input: {
+			account: accounts.targetAccount,
+			networkIdentifier,
+			coSigners: [
+				accounts.mandatoryOne,
+				accounts.mandatoryTow,
+				accounts.optionalOne,
+				accounts.optionalTwo,
+			],
+			transaction: tx,
+		},
+		output: tx,
+	};
+};
+
+const generateValidMultisignatureRegistrationOnlyOptionalMembersTransaction = () => {
+	// basic transaction
+	const tx = {
+		senderPublicKey:
+			'0b211fce4b615083701cb8a8c99407e464b2f9aa4f367095322de1b77e5fcfbe',
+		timestamp: 77045780,
+		type: 12,
+		asset: {
+			mandatoryKeys: [],
+			optionalKeys: [
+				'57df5c3811961939f8dcfa858c6eaefebfaa4de942f7e703bf88127e0ee9cca4',
+				'fa406b6952d377f0278920e3eb8da919e4cf5c68b02eeba5d8b3334fdc0369b6',
+			],
+			numberOfSignatures: 1,
+		},
+		signatures: [],
+	};
+
+	sortKeysDescending(tx.asset.mandatoryKeys);
+	sortKeysDescending(tx.asset.optionalKeys);
+
+	let txBuffer = serializeBasicProperties(tx);
+
+	// Sender signs
+	tx.signatures.push(
+		createSignatureObject(txBuffer, accounts.targetAccount).signature,
+	);
+	// Members sign in order
+	tx.signatures.push(
+		createSignatureObject(txBuffer, accounts.optionalOne).signature,
+	);
+	tx.signatures.push(
+		createSignatureObject(txBuffer, accounts.optionalTwo).signature,
+	);
+	txBuffer = serializeMemberSignatures(tx, txBuffer);
+
+	const id = getId(txBuffer);
+
+	tx.id = id;
+
+	return {
+		input: {
+			account: accounts.targetAccount,
+			networkIdentifier,
+			coSigners: [
+				accounts.mandatoryOne,
+				accounts.mandatoryTow,
+				accounts.optionalOne,
+				accounts.optionalTwo,
+			],
+			transaction: tx,
+		},
+		output: tx,
+	};
+};
+
+const generateValidMultisignatureRegistrationOnlyMandatoryMembersTransaction = () => {
+	// basic transaction
+	const tx = {
+		senderPublicKey:
+			'0b211fce4b615083701cb8a8c99407e464b2f9aa4f367095322de1b77e5fcfbe',
+		timestamp: 77045780,
+		type: 12,
+		asset: {
+			mandatoryKeys: [
+				'4a67646a446313db964c39370359845c52fce9225a3929770ef41448c258fd39',
+				'f1b9f4ee71b5d5857d3b346d441ca967f27870ebee88569db364fd13e28adba3',
+			],
+			optionalKeys: [],
+			numberOfSignatures: 2,
+		},
+		signatures: [],
+	};
+
+	sortKeysDescending(tx.asset.mandatoryKeys);
+	sortKeysDescending(tx.asset.optionalKeys);
+
+	let txBuffer = serializeBasicProperties(tx);
+
+	// Sender signs
+	tx.signatures.push(
+		createSignatureObject(txBuffer, accounts.targetAccount).signature,
+	);
+	// Members sign in order
+	tx.signatures.push(
+		createSignatureObject(txBuffer, accounts.optionalOne).signature,
+	);
+	tx.signatures.push(
+		createSignatureObject(txBuffer, accounts.optionalTwo).signature,
```

Comment:
```suggestion
		createSignatureObject(txBuffer, accounts.mandatoryOne).signature,
```

Reply:
Had to also fix typo in key name so was not able to commit sorry :)